src/lib/encoder.ts

```diff
@@ -74,23 +74,58 @@ const encodeDataSourceWithHash = (
   const verificationMethod = getVerificationMethod(
     verification?.method || UNKNOWN_VERIFICATION_METHOD,
   );
-  return (
-    (verificationMethod
-      ? keccak256(verificationMethod.name).slice(0, 10)
-      : padLeft(0, 8)) +
-    stripHexPrefix(verification ? verification.data : padLeft(0, 64)) +
-    stripHexPrefix(utf8ToHex(dataSource))
-  );
+  return [
+    padLeft(0, 4),
+    stripHexPrefix(
+      verificationMethod
+        ? padLeft(keccak256(verificationMethod.name).slice(0, 10), 8)
+        : padLeft(0, 8),
+    ),
+    stripHexPrefix(
+      verification?.data
+        ? padLeft(verification.data.slice(2).length / 2, 4)
+        : padLeft(0, 4),
```

Comment:
Similar to what @CJ42 suggested previously maybe it makes sense to replace such entries with string literals like `"0000"`?

```suggestion
    '0000',
    stripHexPrefix(
      verificationMethod
        ? padLeft(keccak256(verificationMethod.name).slice(0, 10), 8)
        : '00000000',
    ),
    stripHexPrefix(
      verification?.data
        ? padLeft(verification.data.slice(2).length / 2, 4)
        : '0000',
```